File: credit-check-service/src/main/java/com/senacor/innolab/graalvm/integration/customerservice/model/Customer.java
package com.senacor.innolab.graalvm.integration.customerservice.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import io.quarkus.runtime.annotations.RegisterForReflection;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDate;

@JsonIgnoreProperties(ignoreUnknown = true)
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@RegisterForReflection
public class Customer {

    private Long id;
    private String firstName;
    private String lastName;
    private LocalDate dateOfBirth;
    private BigDecimal income;

}

File: credit-details-service/src/main/kotlin/com/senacor/innolab/graalvm/credit/details/integration/customer/Customer.kt
package com.senacor.innolab.graalvm.credit.details.integration.customer

import io.micronaut.core.annotation.Introspected
import java.math.BigDecimal
import java.time.LocalDate

@Introspected
data class Customer(
    val id: Long,
    val firstName: String?,
    val lastName: String,
    val dateOfBirth: LocalDate,
    val income: BigDecimal
) {
    fun getName() = if (!firstName.isNullOrEmpty()) "$firstName $lastName" else lastName
}